prog/mutation.go

```diff
@@ -374,6 +374,30 @@ func (t *BufferType) mutate(r *randGen, s *state, arg Arg, ctx ArgCtx) (calls []
 	case BufferText:
 		data := append([]byte{}, a.Data()...)
 		a.data = r.mutateText(t.Text, data)
+	case BufferCompressed:
+		data := a.Data()
+		if len(data) == 0 {
+			return
+		}
+		data, err := Decompress(data)
+		if err != nil {
+			panic(fmt.Sprintf("could not decompress data: %v", err))
+		}
+		if len(data) == 0 {
+			return // Do not mutate empty data.
+		}
+		hm := MakeGenericHeatmap(data)
+		// At least one mutation, up to about one mutation every 128 KB of heatmap size.
+		numMutations := r.Intn(hm.Size()/(1<<17)+1) + 1
```

Comment:
What are the typical sizes of metadata for the images we have?
Can't it happen that we have <128K of metadata in most images?
On the second thought, if we have an image with small metadata (exfat), the current algorithm will always do only 1 mutation. But even for "small" images it still makes sense to do several mutations. We could do N mutations per X KB of metadata.

Reply:
Discussed offline, and issue #3530 created.